### PR TITLE
pythonPackages.aioextensions: init at 20.11.1517005

### PR DESCRIPTION
**Pull request by kamadorueda**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kamadorueda** [comment]
```
# Linux kamado 5.4.0-53-generic #59-Ubuntu SMP Wed Oct 21 09:38:44 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
$ git worktree add /home/kamado/.cache/nixpkgs-review/rev-458a7ced196c73854d1fe9a8b3766275d70537e1/nixpkgs ee9dabbdcc465cd28a7c282bda240aaaa224dc08
Preparing worktree (detached HEAD ee9dabbdcc4)
Updating files: 100% (23192/23192), done.
HEAD is now at ee9dabbdcc4 Merge pull request #103806 from r-ryantm/auto-update/python3.7-sphinxcontrib-katex
$ nix-env -f /home/kamado/.cache/nixpkgs-review/rev-458a7ced196c73854d1fe9a8b3766275d70537e1/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit 458a7ced196c73854d1fe9a8b3766275d70537e1
Automatic merge went well; stopped before committing as requested
$ nix-env -f /home/kamado/.cache/nixpkgs-review/rev-458a7ced196c73854d1fe9a8b3766275d70537e1/nixpkgs -qaP --xml --out-path --show-trace --meta
2 packages added:
python37Packages.aioextensions (init at 20.9.2315218) python38Packages.aioextensions (init at 20.9.2315218)

$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/kamado/.cache/nixpkgs-review/rev-458a7ced196c73854d1fe9a8b3766275d70537e1/build.nix
[3 built, 0.0 MiB DL]
2 packages built:
python37Packages.aioextensions python38Packages.aioextensions

$ nix-shell /home/kamado/.cache/nixpkgs-review/rev-458a7ced196c73854d1fe9a8b3766275d70537e1/shell.nix
```

File: pkgs/development/python-modules/aioextensions/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    pytest
+  ];
+
+  meta = with lib; {
+    description = "High performance functions to work with the async IO";
+    homepage = "https://kamadorueda.github.io/aioextensions";
+    license = licenses.mit;
+    maintainers = with maintainers; [
+      kamadorueda
+    ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/aioextensions/default.nix:35]:
```suggestion
    maintainers = with maintainers; [ kamadorueda ];
```

File: pkgs/development/python-modules/aioextensions/default.nix [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  };
+
+  propagatedBuildInputs = [
+    uvloop
+  ];
+
+  checkInputs = [
+    pytest
+  ];
+
```
Comment (jonringer) [inline pkgs/development/python-modules/aioextensions/default.nix:28]:
```suggestion
  checkInputs = [
    pytest
  ];

```

running build_ext

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK

Comment (kamadorueda) [reply pkgs/development/python-modules/aioextensions/default.nix]: now it should be ready

```
nixpkgs$ nix-shell -p python38Packages.aioextensions -I nixpkgs=.
these derivations will be built:
  /nix/store/fk6xk55bl4g0rl1qbncw37mf1y1whw6a-python3.8-aioextensions-20.11.1517005.drv
building '/nix/store/fk6xk55bl4g0rl1qbncw37mf1y1whw6a-python3.8-aioextensions-20.11.1517005.drv'...
Sourcing python-recompile-bytecode-hook.sh
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Using setuptoolsShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing setuptools-check-hook
unpacking sources
unpacking source archive /nix/store/z1664w21d9vb4gx29y5z4g2chqk1ljcg-aioextensions-20.11.1517005.tar.gz
source root is aioextensions-20.11.1517005
setting SOURCE_DATE_EPOCH to timestamp 1605705810 of file aioextensions-20.11.1517005/PKG-INFO
patching sources
configuring
no configure script, doing nothing
building
Executing setuptoolsBuildPhase
running bdist_wheel
running build
running build_py
creating build
creating build/lib
creating build/lib/aioextensions
copying src/aioextensions/__init__.py -> build/lib/aioextensions
copying src/aioextensions/py.typed -> build/lib/aioextensions
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/aioextensions
copying build/lib/aioextensions/__init__.py -> build/bdist.linux-x86_64/wheel/aioextensions
copying build/lib/aioextensions/py.typed -> build/bdist.linux-x86_64/wheel/aioextensions
running install_egg_info
running egg_info
creating src/aioextensions.egg-info
writing src/aioextensions.egg-info/PKG-INFO
writing dependency_links to src/aioextensions.egg-info/dependency_links.txt
writing requirements to src/aioextensions.egg-info/requires.txt
writing top-level names to src/aioextensions.egg-info/top_level.txt
writing manifest file 'src/aioextensions.egg-info/SOURCES.txt'
reading manifest file 'src/aioextensions.egg-info/SOURCES.txt'
writing manifest file 'src/aioextensions.egg-info/SOURCES.txt'
Copying src/aioextensions.egg-info to build/bdist.linux-x86_64/wheel/aioextensions-20.11.1517005-py3.8.egg-info
running install_scripts
adding license file "LICENSE.md" (matched pattern "LICEN[CS]E*")
creating build/bdist.linux-x86_64/wheel/aioextensions-20.11.1517005.dist-info/WHEEL
creating 'dist/aioextensions-20.11.1517005-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'aioextensions/__init__.py'
adding 'aioextensions/py.typed'
adding 'aioextensions-20.11.1517005.dist-info/LICENSE.md'
adding 'aioextensions-20.11.1517005.dist-info/METADATA'
adding 'aioextensions-20.11.1517005.dist-info/WHEEL'
adding 'aioextensions-20.11.1517005.dist-info/top_level.txt'
adding 'aioextensions-20.11.1517005.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
installing
Executing pipInstallPhase
/build/aioextensions-20.11.1517005/dist /build/aioextensions-20.11.1517005
DEPRECATION: The -b/--build/--build-dir/--build-directory option is deprecated. pip 20.3 will remove support for this functionality. A possible replacement is use the TMPDIR/TEMP/TMP environment variable, possibly combined with --no-clean. You can find discussion regarding this at https://github.com/pypa/pip/issues/8333.
Processing ./aioextensions-20.11.1517005-py3-none-any.whl
Requirement already satisfied: uvloop in /nix/store/pd3cvs7hfpw21wzi8wy9ri1m45b1figa-python3.8-uvloop-0.14.0/lib/python3.8/site-packages (from aioextensions==20.11.1517005) (0.14.0)
Installing collected packages: aioextensions
Successfully installed aioextensions-20.11.1517005
/build/aioextensions-20.11.1517005
Finished executing pipInstallPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/2yrqqhwv19hcv3xyp8knhs9fl8x5sms1-python3.8-aioextensions-20.11.1517005
strip is /nix/store/79rshjqnibwcx468m4i31wnmasaj6nf4-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/2yrqqhwv19hcv3xyp8knhs9fl8x5sms1-python3.8-aioextensions-20.11.1517005/lib 
patching script interpreter paths in /nix/store/2yrqqhwv19hcv3xyp8knhs9fl8x5sms1-python3.8-aioextensions-20.11.1517005
checking for references to /build/ in /nix/store/2yrqqhwv19hcv3xyp8knhs9fl8x5sms1-python3.8-aioextensions-20.11.1517005...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
============================= test session starts ==============================
platform linux -- Python 3.8.6, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /build/aioextensions-20.11.1517005
collected 5 items                                                              

test___init__.py .....                                                   [100%]

============================== 5 passed in 5.85s ===============================
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
pytestcachePhase

[nix-shell:~/Documents/kamadorueda/nixpkgs]$ exit
```

**Comment by kamadorueda** [comment]
```
nixpkgs$ nix-review rev HEAD
$ git -c fetch.prune=false fetch --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0
$ git worktree add /home/kamado/.cache/nixpkgs-review/rev-430422ae9e5583bae60f7b78abec0f9458fe5232/nixpkgs 7e25b7113271cf80eb0f950da2e1aa7b382eb268
Preparing worktree (detached HEAD 7e25b711327)
Updating files: 100% (23228/23228), done.
HEAD is now at 7e25b711327 nixos: use nativeBuildInputs in make- iso9660-image and system-tarball
$ nix-env -f /home/kamado/.cache/nixpkgs-review/rev-430422ae9e5583bae60f7b78abec0f9458fe5232/nixpkgs -qaP --xml --out-path --show-trace
$ git merge --no-commit 430422ae9e5583bae60f7b78abec0f9458fe5232
Updating 7e25b711327..430422ae9e5
Fast-forward
 .../python-modules/aioextensions/default.nix   | 35 +++++++++++++++++++
 pkgs/top-level/python-packages.nix             |  2 ++
 2 files changed, 37 insertions(+)
 create mode 100644 pkgs/development/python-modules/aioextensions/default.nix
$ nix-env -f /home/kamado/.cache/nixpkgs-review/rev-430422ae9e5583bae60f7b78abec0f9458fe5232/nixpkgs -qaP --xml --out-path --show-trace --meta
2 packages added:
python37Packages.aioextensions (init at 20.11.1517005) python38Packages.aioextensions (init at 20.11.1517005)

$ nix build --no-link --keep-going --option build-use-sandbox relaxed -f /home/kamado/.cache/nixpkgs-review/rev-430422ae9e5583bae60f7b78abec0f9458fe5232/build.nix
[2 built, 0.0 MiB DL]
2 packages built:
python37Packages.aioextensions python38Packages.aioextensions

$ nix-shell /home/kamado/.cache/nixpkgs-review/rev-430422ae9e5583bae60f7b78abec0f9458fe5232/shell.nix

[nix-shell:~/.cache/nixpkgs-review/rev-430422ae9e5583bae60f7b78abec0f9458fe5232]$
```

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 103819 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>2 packages built:</summary>  <ul>    <li>python37Packages.aioextensions</li>    <li>python38Packages.aioextensions</li>  </ul></details>

**Pull request merged**